Nan examples fail on node 12

With node 12 the nan examples fail:

```
sephsmac:nan josephg$ node --version
v12.1.0
sephsmac:nan josephg$ pwd
/Users/josephg/3rdparty/node-addon-examples/1_hello_world/nan
sephsmac:nan josephg$ node-gyp rebuild
...
  CXX(target) Release/obj.target/hello/hello.o
../hello.cc:9:68: error: too few arguments to function call, single argument 'context' was not specified
               Nan::New<v8::FunctionTemplate>(Method)->GetFunction());
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/josephg/.node-gyp/12.1.0/include/node/v8.h:5947:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/josephg/.node-gyp/12.1.0/include/node/v8config.h:347:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
1 error generated.
make: *** [Release/obj.target/hello/hello.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
```

Other examples fail in similar ways.

Updating to the latest version of nan does not fix the issue.

--- a/.gitignore
+++ b/.gitignore
@@ -4,6 +4,7 @@ Debug/
 Release/
 *.lock
 *.log
+package-lock.json
 npm-debug.log*
 
 .idea/

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,7 +2,7 @@ language: node_js
 node_js:
   - "8"
   - "10"
-#  - "12" enable this when fix issue on Node.js 12
+  - "12"
 cache:
   npm
 before_script:

--- a/1_hello_world/nan/hello.cc
+++ b/1_hello_world/nan/hello.cc
@@ -5,8 +5,9 @@ void Method(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 }
 
 void Init(v8::Local<v8::Object> exports) {
+  v8::Local<v8::Context> context = exports->CreationContext();
   exports->Set(Nan::New("hello").ToLocalChecked(),
-               Nan::New<v8::FunctionTemplate>(Method)->GetFunction());
+               Nan::New<v8::FunctionTemplate>(Method)->GetFunction(context).ToLocalChecked());
 }
 
 NODE_MODULE(hello, Init)

--- a/2_function_arguments/nan/addon.cc
+++ b/2_function_arguments/nan/addon.cc
@@ -1,6 +1,7 @@
 #include <nan.h>
 
 void Add(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
 
   if (info.Length() < 2) {
     Nan::ThrowTypeError("Wrong number of arguments");
@@ -12,16 +13,17 @@ void Add(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     return;
   }
 
-  double arg0 = info[0]->NumberValue();
-  double arg1 = info[1]->NumberValue();
+  double arg0 = info[0]->NumberValue(context).FromJust();
+  double arg1 = info[1]->NumberValue(context).FromJust();
   v8::Local<v8::Number> num = Nan::New(arg0 + arg1);
 
   info.GetReturnValue().Set(num);
 }
 
 void Init(v8::Local<v8::Object> exports) {
+  v8::Local<v8::Context> context = exports->CreationContext();
   exports->Set(Nan::New("add").ToLocalChecked(),
-               Nan::New<v8::FunctionTemplate>(Add)->GetFunction());
+               Nan::New<v8::FunctionTemplate>(Add)->GetFunction(context).ToLocalChecked());
 }
 
 NODE_MODULE(addon, Init)

--- a/3_callbacks/nan/package.json
+++ b/3_callbacks/nan/package.json
@@ -6,7 +6,7 @@
   "private": true,
   "gypfile": true,
   "dependencies": {
-    "bindings": "~1.2.1",
-    "nan": "^2.0.0"
+    "bindings": "~1.5.0",
+    "nan": "^2.14.0"
   }
 }

--- a/4_object_factory/nan/addon.cc
+++ b/4_object_factory/nan/addon.cc
@@ -1,15 +1,17 @@
 #include <nan.h>
 
 void CreateObject(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
   v8::Local<v8::Object> obj = Nan::New<v8::Object>();
-  obj->Set(Nan::New("msg").ToLocalChecked(), info[0]->ToString());
+  obj->Set(Nan::New("msg").ToLocalChecked(), info[0]->ToString(context).ToLocalChecked());
 
   info.GetReturnValue().Set(obj);
 }
 
 void Init(v8::Local<v8::Object> exports, v8::Local<v8::Object> module) {
+  v8::Local<v8::Context> context = exports->CreationContext();
   module->Set(Nan::New("exports").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(CreateObject)->GetFunction());
+      Nan::New<v8::FunctionTemplate>(CreateObject)->GetFunction(context).ToLocalChecked());
 }
 
 NODE_MODULE(addon, Init)

--- a/4_object_factory/nan/package.json
+++ b/4_object_factory/nan/package.json
@@ -6,7 +6,7 @@
   "private": true,
   "gypfile": true,
   "dependencies": {
-    "bindings": "~1.2.1",
-    "nan": "^2.0.0"
+    "bindings": "~1.5.0",
+    "nan": "^2.14.0"
   }
 }

--- a/5_function_factory/nan/addon.cc
+++ b/5_function_factory/nan/addon.cc
@@ -5,8 +5,9 @@ void MyFunction(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 }
 
 void CreateFunction(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
   v8::Local<v8::FunctionTemplate> tpl = Nan::New<v8::FunctionTemplate>(MyFunction);
-  v8::Local<v8::Function> fn = tpl->GetFunction();
+  v8::Local<v8::Function> fn = tpl->GetFunction(context).ToLocalChecked();
 
   // omit this to make it anonymous
   fn->SetName(Nan::New("theFunction").ToLocalChecked());

--- a/6_object_wrap/nan/myobject.cc
+++ b/6_object_wrap/nan/myobject.cc
@@ -9,6 +9,7 @@ MyObject::~MyObject() {
 }
 
 void MyObject::Init(v8::Local<v8::Object> exports) {
+  v8::Local<v8::Context> context = exports->CreationContext();
   Nan::HandleScope scope;
 
   // Prepare constructor template
@@ -21,14 +22,16 @@ void MyObject::Init(v8::Local<v8::Object> exports) {
   Nan::SetPrototypeMethod(tpl, "plusOne", PlusOne);
   Nan::SetPrototypeMethod(tpl, "multiply", Multiply);
 
-  constructor.Reset(tpl->GetFunction());
-  exports->Set(Nan::New("MyObject").ToLocalChecked(), tpl->GetFunction());
+  constructor.Reset(tpl->GetFunction(context).ToLocalChecked());
+  exports->Set(Nan::New("MyObject").ToLocalChecked(),
+      tpl->GetFunction(context).ToLocalChecked());
 }
 
 void MyObject::New(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
   if (info.IsConstructCall()) {
     // Invoked as constructor: `new MyObject(...)`
-    double value = info[0]->IsUndefined() ? 0 : info[0]->NumberValue();
+    double value = info[0]->IsUndefined() ? 0 : info[0]->NumberValue(context).FromJust();
     MyObject* obj = new MyObject(value);
     obj->Wrap(info.This());
     info.GetReturnValue().Set(info.This());
@@ -37,7 +40,6 @@ void MyObject::New(const Nan::FunctionCallbackInfo<v8::Value>& info) {
     const int argc = 1;
     v8::Local<v8::Value> argv[argc] = { info[0] };
     v8::Local<v8::Function> cons = Nan::New<v8::Function>(constructor);
-    v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
     info.GetReturnValue().Set(
         cons->NewInstance(context, argc, argv).ToLocalChecked());
   }
@@ -55,15 +57,15 @@ void MyObject::PlusOne(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 }
 
 void MyObject::Multiply(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
   MyObject* obj = ObjectWrap::Unwrap<MyObject>(info.Holder());
-  double multiple = info[0]->IsUndefined() ? 1 : info[0]->NumberValue();
+  double multiple = info[0]->IsUndefined() ? 1 : info[0]->NumberValue(context).FromJust();
 
   v8::Local<v8::Function> cons = Nan::New<v8::Function>(constructor);
 
   const int argc = 1;
   v8::Local<v8::Value> argv[argc] = { Nan::New(obj->value_ * multiple) };
 
-  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
   info.GetReturnValue().Set(
       cons->NewInstance(context, argc, argv).ToLocalChecked());
 }

--- a/7_factory_wrap/nan/addon.cc
+++ b/7_factory_wrap/nan/addon.cc
@@ -6,12 +6,14 @@ void CreateObject(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 }
 
 void InitAll(v8::Local<v8::Object> exports, v8::Local<v8::Object> module) {
+  v8::Local<v8::Context> context = exports->CreationContext();
+
   Nan::HandleScope scope;
 
   MyObject::Init();
 
   module->Set(Nan::New("exports").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(CreateObject)->GetFunction());
+      Nan::New<v8::FunctionTemplate>(CreateObject)->GetFunction(context).ToLocalChecked());
 }
 
 NODE_MODULE(addon, InitAll)

--- a/7_factory_wrap/nan/myobject.cc
+++ b/7_factory_wrap/nan/myobject.cc
@@ -19,12 +19,14 @@ void MyObject::Init() {
   tpl->PrototypeTemplate()->Set(Nan::New("plusOne").ToLocalChecked(),
       Nan::New<v8::FunctionTemplate>(PlusOne));
 
-  constructor.Reset(tpl->GetFunction());
+  constructor.Reset(tpl->GetFunction(Nan::GetCurrentContext()).ToLocalChecked());
 }
 
 void MyObject::New(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
+
   MyObject* obj = new MyObject();
-  obj->counter_ = info[0]->IsUndefined() ? 0 : info[0]->NumberValue();
+  obj->counter_ = info[0]->IsUndefined() ? 0 : info[0]->NumberValue(context).FromJust();
   obj->Wrap(info.This());
 
   info.GetReturnValue().Set(info.This());

--- a/8_passing_wrapped/nan/addon.cc
+++ b/8_passing_wrapped/nan/addon.cc
@@ -8,20 +8,24 @@ void CreateObject(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 }
 
 void Add(const Nan::FunctionCallbackInfo<v8::Value>& info) {
-  MyObject* obj1 = Nan::ObjectWrap::Unwrap<MyObject>(info[0]->ToObject());
-  MyObject* obj2 = Nan::ObjectWrap::Unwrap<MyObject>(info[1]->ToObject());
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
+
+  MyObject* obj1 = Nan::ObjectWrap::Unwrap<MyObject>(info[0]->ToObject(context).ToLocalChecked());
+  MyObject* obj2 = Nan::ObjectWrap::Unwrap<MyObject>(info[1]->ToObject(context).ToLocalChecked());
   double sum = obj1->Val() + obj2->Val();
   info.GetReturnValue().Set(Nan::New(sum));
 }
 
 void InitAll(v8::Local<v8::Object> exports) {
+  v8::Local<v8::Context> context = exports->CreationContext();
+
   MyObject::Init();
 
   exports->Set(Nan::New("createObject").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(CreateObject)->GetFunction());
+      Nan::New<v8::FunctionTemplate>(CreateObject)->GetFunction(context).ToLocalChecked());
 
   exports->Set(Nan::New("add").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(Add)->GetFunction());
+      Nan::New<v8::FunctionTemplate>(Add)->GetFunction(context).ToLocalChecked());
 }
 
 NODE_MODULE(addon, InitAll)

--- a/8_passing_wrapped/nan/myobject.cc
+++ b/8_passing_wrapped/nan/myobject.cc
@@ -14,12 +14,13 @@ void MyObject::Init() {
   tpl->SetClassName(Nan::New("MyObject").ToLocalChecked());
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
 
-  constructor.Reset(tpl->GetFunction());
+  constructor.Reset(tpl->GetFunction(Nan::GetCurrentContext()).ToLocalChecked());
 }
 
 void MyObject::New(const Nan::FunctionCallbackInfo<v8::Value>& info) {
+  v8::Local<v8::Context> context = info.GetIsolate()->GetCurrentContext();
   MyObject* obj = new MyObject();
-  obj->val_ = info[0]->IsUndefined() ? 0 : info[0]->NumberValue();
+  obj->val_ = info[0]->IsUndefined() ? 0 : info[0]->NumberValue(context).FromJust();
   obj->Wrap(info.This());
 
   info.GetReturnValue().Set(info.This());

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 Node.js Addon Examples
 =========================================
-[![Build Status](https://travis-ci.org/nodejs/node-addon-examples.svg?branch=master)](https://travis-ci.org/nodejs/node-addon-examples)
+[![Build Status](https://travis-ci.com/nodejs/node-addon-examples.svg?branch=master)](https://travis-ci.com/nodejs/node-addon-examples)
 
 **A repository of [Node.js Addons](https://nodejs.org/api/addons.html#addons_c_addons) examples.**
 

--- a/async_pi_estimate/nan/addon.cc
+++ b/async_pi_estimate/nan/addon.cc
@@ -3,7 +3,6 @@
 #include "async.h"  // NOLINT(build/include)
 
 using v8::FunctionTemplate;
-using v8::Handle;
 using v8::Object;
 using v8::String;
 using Nan::GetFunction;

--- a/async_pi_estimate/nan/sync.cc
+++ b/async_pi_estimate/nan/sync.cc
@@ -5,7 +5,7 @@
 // Simple synchronous access to the `Estimate()` function
 NAN_METHOD(CalculateSync) {
   // expect a number as the first argument
-  int points = info[0]->Uint32Value();
+  int points = info[0]->Uint32Value(Nan::GetCurrentContext()).FromJust();
   double est = Estimate(points);
 
   info.GetReturnValue().Set(est);

